internal/syscall/src/lib.rs

```diff
@@ -235,13 +238,41 @@ pub trait SyscallHandler: AddressValidator + Sized {
     fn write(&mut self, fd: libc::c_int, buf: UntrustedRef<u8>, count: libc::size_t) -> Result {
         let buf = buf.validate_slice(count, self).ok_or(libc::EFAULT)?;
 
-        let c = self.new_cursor();
-        let (_, buf) = c.copy_from_slice(buf.as_ref()).or(Err(libc::EMSGSIZE))?;
-        let buf = buf.as_ptr();
-        let host_virt = self.translate_shim_to_host_addr(buf);
+        let mut to_write = count;
+        let mut written = 0;
 
-        let ret = unsafe { self.proxy(request!(libc::SYS_write => fd, host_virt, count))? };
-        Ok(ret)
+        loop {
+            let count = usize::min(to_write, Block::buf_capacity());
```

Comment:
Why do we need to specify this twice?

Reply:
This shadows the `count` parameter which is immutable. If he hadn't shadowed it, this wouldn't compile unless he made the parameter mutable which is not desirable.

Reply:
Ahh, sorry, I totally spaced out on the fact that the other `count` was in `read()` and not `write()`. Makes sense now!